.ci/metrics/metrics.py

```diff
@@ -81,6 +90,156 @@ class GaugeMetric:
     time_ns: int
 
 
+@dataclass
+class AggregateMetric:
+    aggregate_name: str
+    aggregate_queue_time: int
+    aggregate_run_time: int
+    aggregate_status: int
+    completed_at_ns: int
+    workflow_id: int
+
+
+def _construct_aggregate(ag_name: str, job_list: list[JobMetrics]) -> AggregateMetric:
+    """Create a libc++ AggregateMetric from a list of libc++ JobMetrics
+
+    How aggregates are computed:
+    queue time: Time from when first job in group is created until last job in
+                group has started.
+    run time: Time from when first job in group starts running until last job
+              in group finishes running.
+    status: logical 'and' of all the job statuses in the group.
+
+    Args:
+      ag_name: The name for this particular AggregateMetric
+
+      job_list: This list of JobMetrics to be combined into the AggregateMetric.
+        The input list should contain all (and only!) the libc++ JobMetrics
+        for a particular stage and a particular workflow_id.
+
+    Returns:
+      Returns the AggregateMetric constructed from the inputs.
+    """
+
+    # Initialize the aggregate values
+    earliest_create = job_list[0].created_at_ns
+    earliest_start = job_list[0].started_at_ns
+    earliest_complete = job_list[0].completed_at_ns
+    latest_start = job_list[0].started_at_ns
+    latest_complete = job_list[0].completed_at_ns
+    ag_status = job_list[0].status
+    ag_workflow_id = job_list[0].workflow_id
+
+    # Go through rest of jobs for this workflow id, if any, updating stats
+    if len(job_list) > 1:
+        for job in job_list[1:]:
+            # Update the status
+            ag_status = ag_status and job.status
+            # Get the earliest & latest times
+            if job.created_at_ns < earliest_create:
+                earliest_create = job.created_at_ns
+            if job.completed_at_ns < earliest_complete:
+                earliest_complete = job.completed_at_ns
+            if job.started_at_ns > latest_start:
+                latest_start = job.started_at_ns
+            if job.started_at_ns < earliest_start:
+                earliest_start = job.started_at_ns
+            if job.completed_at_ns > latest_complete:
+                latest_complete = job.completed_at_ns
+
+    # Compute aggregate run time (in seconds, not ns)
+    ag_run_time = (latest_complete - earliest_start) / 1000000000
+    # Compute aggregate queue time (in seconds, not ns)
+    ag_queue_time = (latest_start - earliest_create) / 1000000000
+    # Append the aggregate metrics to the workflow metrics list.
+    aggregate = AggregateMetric(ag_name, ag_queue_time, ag_run_time, ag_status,
```

Comment:
Nit: Drop the variable creation and just return directly?

Reply:
Done.